src/session.c

```diff
@@ -587,15 +591,24 @@ static int construct_resolve_slist(CURL *session, bool force) {
         filesystem_domain, filesystem_port);
     res = getaddrinfo(filesystem_domain, filesystem_port, &hints, &aihead);
```

Comment:
It doesn't seem like we should call this `res` any more.

Maybe the next line should just be `if(getaddrinfo(filesystem_domain, filesystem_port, &hints, &aihead)) {`
?